src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -71,6 +72,22 @@ public void setFrameworkId(FrameworkID frameworkId) throws InterruptedException,
     zkState.store(value).get();
   }
 
+  public void resetDeadNodeTimeStamps() {
+    Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+
+    if (getDeadJournalNodes().size() > 0) {
+      deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
After previous discussion, the conclusion was that we need a TODO to remove timeout for datanodes, but for NN and JN's one timeout per node type should be sufficient. I'm not convinced we need to add complexity to the code. Will follow up with you on this and we can discuss in person.